portals/portals.c

```diff
@@ -107,11 +108,13 @@ int PtlInit(void)
 
 void PtlFini(void)
 {
-	struct linked_list *current = dev_list;
-	while (current->next != NULL) {
-		struct linked_list *n = current->next;
-		free(current);
-		current = n;
+	if(dev_list != NULL) {
+		struct linked_list *current = dev_list;
+		while (current->next != NULL) {
+			struct linked_list *n = current->next;
+			free(current);
+			current = n;
+		}
 	}
 	return swptl_func_libfini(ctx_global);

```

Comment:
Should we wait for swptl_func_libfini() success before freeing the swptl_dev list ?